pkg/sql/information_schema.go

```diff
@@ -525,6 +542,9 @@ CREATE TABLE information_schema.tables (
 );`,
 	populate: func(ctx context.Context, p *planner, prefix string, addRow func(...tree.Datum) error) error {
 		return forEachTableDesc(ctx, p, prefix, func(db *sqlbase.DatabaseDescriptor, table *sqlbase.TableDescriptor) error {
+			if table.IsSequence() {
```

Comment:
No super strong feelings but it feels to me like this might merit inverting and extracting the logic somehow, like `IsRegularTable`? Or just something so that this table code doesn't have to consider the special case of sequences explicitly (or maybe since sequences are such a special case it's fine, I'm not sure).

Reply:
I see your point — `IsTable` does that, but in `information_schema.tables` we actually don't want that, since it also returns views.

If we had a few more cases like this I'd do a refactor, but it doesn't seem that bad now. As refactors go, I think creating an enum for descriptor types and replacing if/else trees with switch statements would get more mileage.